.golangci.yml

```diff
@@ -44,6 +44,8 @@ linters:
   - whitespace
 
 linters-settings:
+  gocyclo:
+    min-complexity: 50
```

Comment:
I originally suggested to do this to @ykakarap , although not sure at this point if it makes sense to keep it around; reducing the number of if branches or lines of code in a particular function or method isn't a particularly useful target. We should always strive for readability and simplicity rather than refactoring based on a arbitrary number.

Reply:
>Is the previous level 30? (default from https://golangci-lint.run/usage/linters/#gocyclo)

Yes. 

I am +1 to dropping it. And looks like Vince is also okay with dropping it: https://github.com/kubernetes-sigs/cluster-api/pull/6183#discussion_r812047541

Reply:
I'm fine with dropping or upping the limit -  agreed it's not the most useful linter, but it does work as a sort of a low pass filter for complexity. @sbueringer @fabriziopandini  WDYT?

Reply:
I'm fine with dropping it given that our most complex code is not even found by gocyclo and it's mostly finding relatively straightforward code right now.

Reply:
Will drop it.